libs/server-sdk/src/config/builders/data_system/data_system_builder.cpp

```diff
@@ -0,0 +1,50 @@
+#include <launchdarkly/server_side/config/builders/data_system/data_system_builder.hpp>
+#include "defaults.hpp"
+
+namespace launchdarkly::server_side::config::builders {
+
+DataSystemBuilder::DataSystemBuilder()
+    : method_builder_(std::nullopt), config_(Defaults::DataSystemConfig()) {}
+
+DataSystemBuilder& DataSystemBuilder::Method(BackgroundSync bg_sync) {
+    method_builder_ = std::move(bg_sync);
+    return *this;
+}
+
+DataSystemBuilder& DataSystemBuilder::Method(LazyLoad lazy_load) {
+    method_builder_ = std::move(lazy_load);
+    return *this;
+}
+
+DataSystemBuilder& DataSystemBuilder::Disabled(bool const disabled) {
+    config_.disabled = disabled;
+    return *this;
+}
+
+tl::expected<built::DataSystemConfig, Error> DataSystemBuilder::Build() const {
```

Comment:
This is pretty simple in concept but just gross to look at. 

The problem is unifying the return values of `LazyLoad::Build` (fallible, `tl::expected`) and `BackgroundSync::Build` (not fallible, since if no source is provided we default to streaming.)

Instead of doing it in build, could instead call `Build` in the `::Method` implementations. Then we'd need to store an:
```c++
std::variant<
  tl::expected<LazyLoadConfig, Error>, 
  tl::expected<BackgroundSyncConfig, Error>
>
```
Maybe not horrible 🤷